.github/workflows/docker-release.yaml

```diff
@@ -0,0 +1,65 @@
+name: Build and push release images
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Login to DockerHub
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+      - name: Extract tag name
+        id: extract_shortref
+        run: echo "##[set-output name=shortref;]$(echo ${GITHUB_REF#refs/tags/})"
+      - name: Build binary
+        env:
+          TAG: ${{ steps.extract_shortref.outputs.shortref }}
+          COMMIT: ${{ github.sha }}
+          REPO_INFO: https://github.com/${{ github.repository }}.git
+        id: docker_build_binary
+        uses: docker/build-push-action@v2
```

Comment:
Like said in the `Makefile` comment, I recommend that we use the makefile (instead of this action) to build the image, to ensure that the exact same build method is used by CI and manual workflows. This action can be used for the push later.

Reply:
Moving to Docker's action was partially to benefit from upstream work and partially because its build caching just works--the Go binary build time is far longer than the other stages. Not sure if that's easily doable with the Makefile alone.

The Makefile target doesn't do anything other than populate basic values for the `docker build` invocation, and we can gather the same info/store it in the environment equally well with the action, so I don't see a concern for the difference in the manual workflow.

Reply:
use buildkit and you can benefit from `RUN --mount=type=cache,target=/root/.m2 mvn -o install` for cacheing purposes

[prior art](https://github.com/Kong/docker-pulp/blob/main/pulp-core/Dockerfile#L40)

I agree with the sentiment of having the local way of doing things being equivallent to the CI way of doing thing. An option would be to have the logic in github actions and tell Make to use it

```
.github/workflows/docker_build.yml
.github/workflows/docker_release.yml

docker_build:
    act -W .github/workflows/docker_build.yml -v
```